### PR TITLE
[FLOC-1955] Rename slaves.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-1955

In addition to the changes suggested in the issue, this makes the rest of slave names more consistent.

# TODO
- [x] Add `aws` prefix.
- [x] Use different separator for the slave index.
- [x] Documentation of the consistency.

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/build.clusterhq.com/94)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -165,6 +165,18 @@ The buildslave is constructed with a ``pip.conf`` file that points at https://s3
 
 .. [1] Create credentials at https://console.aws.amazon.com/iam/home#users.
 
+Slaves
+======
+
+Naming
+------
+
+Slaves are named with a number of components, separated by ``\``.
```
Comment (wallrj) [inline README.rst]: ``/``

File: README.rst [hunk 2]
```diff
@@ -165,6 +165,18 @@ The buildslave is constructed with a ``pip.conf`` file that points at https://s3
...
+The primary component is the name of the operating system running on the slave (e.g. ``fedora-20``).
+There is usually a prefix indicating where the slave is hosted (e.g. ``aws`` or ``redhat-openstack``).
+If there is an unusual configuration to the slave, there is a tag describing it (e.g. ``zfs-head``).
+There is usually a numberical suffix indicating which instance of similarily configured slaves this is.
```
Comment (wallrj) [inline README.rst]: typo "numerical"

Comment (wallrj) [inline README.rst]: typo "similarily"

File: config.py [hunk 3]
```diff
@@ -148,8 +148,8 @@ def get_cloud_init(name, base, password, provider, privateData, slavePortnum):
             )
             c['slaves'].append(slave)
     else:
-        for i, password in enumerate(slaveConfig['passwords']):
-            name = '%s-%d' % (base, i)
+        for index, password in enumerate(slaveConfig['passwords']):
```
Comment (wallrj) [inline config.py]: I suppose we could auto-generate the passwords for the non on-demand slaves too.

File: README.rst [hunk 4]
```diff
@@ -252,55 +264,55 @@ To configure any Fedora 20 bare metal machine (e.g. on OnMetal as above)::
...
         HostName <novahost_public_hostname_or_ip_address>
-        ProxyCommand ssh pistoncloud-jumphost nc %h %p
+        ProxyCommand ssh redhat-openstack-jumphost nc %h %p
 
 With that ``~/.ssh/config`` content in place, run:
 
 .. code-block:: console
 
-   [laptop] $ fab -f slave/pistoncloud/fabfile.py create_server:clusterhq_joebloggs
+   [laptop] $ fab -H redhat-openstack-novahost -f slave/redhat-openstack/fabfile.py create_server:clusterhq_joebloggs
```
Comment (wallrj) [inline README.rst]: :question: Fine :smile: I was going to say "I don't know why you prefer this over hard coding the hosts in `~/.ssh/config`." it seems more error prone, but i suppose at least this way I can override the destination host if a need to.

**Comment by wallrj** [comment]
Thanks @tomprince 

 * I skimmed through the documentation changes and noted a couple of typos which can be fixed later.
 * The list of build slaves is nice and consistent http://build.clusterhq.com/buildslaves

I'll run some builds today and let you know if there are any problems.

**Comment by wallrj** [comment]
I notice that `flocker/functional/redhat-openstack/centos-7/storage-driver` is no longer listed in the `expected_failures` section, although it seems to be configured that way in lastpass. 
Perhaps the build master needs restarting again.

**Comment by tomprince** [comment]
> I notice that flocker/functional/redhat-openstack/centos-7/storage-driver is no longer listed in the expected_failures section, although it seems to be configured that way in lastpass.
Perhaps the build master needs restarting again.

Yep. I wasn't expecting anybody to be working today, and had other stuff I was planning to land, so I was just going to wait until I restarted it for other reasons.

File: config.py (hunk 3 above)
Comment (tomprince) [inline config.py]: Not dynamically. They need to be communicated to the slave out-of-band, and thus a record needs to be kept of them somewhere. Certainly they should be originally created randomly.

File: README.rst (hunk 4 above)
Comment (tomprince) [inline README.rst]: I do have them hard-coded in `~/.ssh/config`, but with different names than are suggested here.

A better long-term option might be to get fabric to use a custom ssh-config that can be stored in last-pass, perhaps.

**Pull request merged**